airflow/plugins/operators/scrape_ntd_api.py

```diff
@@ -0,0 +1,125 @@
+# import os
+import gzip
+import logging
+from typing import ClassVar, List  # , Optional
+
+import pandas as pd  # type: ignore
+import pendulum
+import requests
+
+# import typer
+from calitp_data_infra.storage import PartitionedGCSArtifact, get_fs  # type: ignore
+from pydantic import HttpUrl, parse_obj_as
+
+from airflow.models import BaseOperator  # type: ignore
+
+# Restore for prod
+# API_BUCKET = os.environ["CALITP_BUCKET__NTD_API_DATA_PRODUCTS"]
+
+API_BUCKET = "gs://calitp-ntd-api-products"
+
+
+class NtdDataProductAPIExtract(PartitionedGCSArtifact):
+    bucket: ClassVar[str]
+    year: str
+    product: str
+    execution_ts: pendulum.DateTime = pendulum.now()
+    dt: pendulum.Date = execution_ts.date()
+    root_url: str
+    endpoint_id: str
+    file_format: str
+    partition_names: ClassVar[List[str]] = ["dt", "execution_ts"]
+
+    @property
+    def table(self) -> str:
+        return self.product
+
+    @property
+    def filename(self) -> str:
+        return self.table
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def fetch_from_ntd_api(self):
+        """ """
+
+        logging.info(f"Downloading NTD data for {self.year} / {self.product}.")
+
+        try:
+            url = (
+                self.root_url + self.endpoint_id + self.file_format + "?$limit=5000000"
+            )
+
+            validated_url = parse_obj_as(HttpUrl, url)
+
+            response = requests.get(validated_url).content
+
+            if response is None or len(response) == 0:
+                logging.info(
+                    f"There is no data to download for {self.year} / {self.product}. Ending pipeline."
+                )
+
+                pass
+            else:
+                logging.info(
+                    f"Downloaded {self.product} data for {self.year} with {len(response)} rows!"
+                )
```

Comment:
**suggestion (non-blocking):** We should probably raise an exception when there is a non-200 response from the API (e.g., a 404 response will still return some JSON, but not in the structure that's valid for the table).

This can be done in a follow-on issue.

Reply:
caputured in this ticket:

https://github.com/cal-itp/data-infra/issues/3474